userspace/libsinsp/k8s_handler.cpp

```diff
@@ -166,8 +166,6 @@ void k8s_handler::make_http()
 		m_handler->set_path(m_path);
 		m_handler->set_id(m_id);
 		m_collector->set_steady_state(true);
-		m_watching = true;
-		m_blocking_socket = false;
```

Comment:
Probably I would set them to the right value instead of removing them, just to be clearer and future-proof:thinking: So something like 

```c
m_watching = false;
m_blocking_socket = true;
```

Reply:
These two variables are set when the handler switches to the `watching mode`. I moved them out of `k8s_handler::make_http()` function and put them in the `k8s_handler::check_state()`. I want to make sure that the variables are set only when we switch to the `watch mode` that happens by calling the `k8s_handler::check_state()` function. You can easily see that `k8s_handler::make_http` is called in other places making it unsuitable to set those variables.